src/lib/transition_router/transition_router.ml

```diff
@@ -106,15 +106,10 @@ module Make (Inputs : Inputs_intf) = struct
     transition_writer_ref := bootstrap_controller_writer ;
     Transition_frontier.close frontier ;
     Broadcast_pipe.Writer.write frontier_w None |> don't_wait_for ;
-    let ask_best_tip_signal = Ivar.create () in
-    (* We make sure that we are connected to somebody before we eagerly bootstrap *)
-    don't_wait_for
-      (let%map () = Ivar.read (Network.first_connection network) in
-       Ivar.fill ask_best_tip_signal ()) ;
     upon
       (Bootstrap_controller.run ~logger ~trust_system ~verifier ~network
-         ~ledger_db ~frontier ~transition_reader:!transition_reader_ref
-         ~ask_best_tip_signal) (fun (new_frontier, collected_transitions) ->
+         ~ledger_db ~frontier ~transition_reader:!transition_reader_ref)
```

Comment:
I'm pretty sure that you want to stick the `let%bind () = Ivar.read (Network.first_connection network) in` above this now that you don't have the Ivar you are threading through.